File: apps/holtburger-3d/src/lib/assets/material-variants.ts
export const BASE_MATERIAL_VARIANT_SIGNATURE = "base";
export const LEGACY_SAMPLER_CLAMP_MATERIAL_VARIANT_SIGNATURE = "sampler=clamp";
export const LEGACY_SAMPLER_REPEAT_MATERIAL_VARIANT_SIGNATURE =
	"sampler=repeat";

export type LegacySamplerMaterialVariant = "clamp" | "repeat";

export function formatLegacySamplerMaterialVariantSignature(
	variant: LegacySamplerMaterialVariant,
): string {
	return variant === "repeat"
		? LEGACY_SAMPLER_REPEAT_MATERIAL_VARIANT_SIGNATURE
		: LEGACY_SAMPLER_CLAMP_MATERIAL_VARIANT_SIGNATURE;
}

File: crates/holtburger-content/src/material_variants.rs
pub const LEGACY_SAMPLER_CLAMP_MATERIAL_VARIANT_SIGNATURE: &str = "sampler=clamp";
pub const LEGACY_SAMPLER_REPEAT_MATERIAL_VARIANT_SIGNATURE: &str = "sampler=repeat";

pub fn legacy_sampler_material_variant_signature(repeats: bool) -> &'static str {
    if repeats {
        LEGACY_SAMPLER_REPEAT_MATERIAL_VARIANT_SIGNATURE
    } else {
        LEGACY_SAMPLER_CLAMP_MATERIAL_VARIANT_SIGNATURE
    }
}
